src/app/App.tsx

```diff
@@ -32,6 +38,15 @@ const App: React.FC<AppProps> = ({
   const { tokenRefreshing, tokenVerifying } = useAuth();
   const loading = tokenRefreshing || tokenVerifying || willRedirect;
 
+  useEffect(() => {
+    const attachClient = async () => {
+      const { apolloClient } = await getSaleorApi();
+      if (window) window.__APOLLO_CLIENT__ = apolloClient;
+    };
+    if (process.env.NEXT_PUBLIC_ENABLE_APOLLO_DEVTOOLS === "true")
+      attachClient();
+  }, []);
```

Comment:
This code should be outside any React component as it's not bound to UI in any form.